packages/core/src/new-api/internal/execution/execution-engine.ts

```diff
@@ -80,6 +85,13 @@ export class ExecutionEngine {
 
     await this._checkNonces(state, futures);
 
+    // Record start of the run (which resets timeout to started)
+    const startRun: StartRunMessage = {
+      type: "run-start",
+    };
+
+    await this._apply(state, startRun);
```

Comment:
After the nonces check we should be good to start an execution. At that point we record a "run-start" message, which resets `TIMEOUT` futures to `STARTED`. The execution engine will jump those futures to same checks as before the timeout on the previous run.

NOTE: we should probably enhance the start run message to include chainId, to better support a chainId check in reconciliation.

Reply:
Do we need a message for this? Shouldn't the normal reducer flow reset TIMEOUT to STARTED?

Not opposed to having it, just trying to understand the reasoning. I do see one advantage: doing it this way the normal reducer creates the state as seen in the journal, and then we decide what to do with timeout as part of the execution. This is handy for things like displaying the current status in plan, whereas switching the TIMEOUTs there wouldn't allow us to do it.

Reply:
Well, I think I answered myself.

Reply:
I want to make that message richer as well: chainId and a count of runs. chainId for reconciliation purposes (you can't run on a different chain from previous run) and count of runs to improve both UI and UX of understanding the journal as a log.